realm-transformer/src/main/kotlin/io/realm/transformer/ByteCodeModifier.kt

```diff
@@ -162,9 +162,16 @@ class BytecodeModifier {
         @Throws(CannotCompileException::class)
         override fun edit(fieldAccess: FieldAccess) {
             logger.debug("      Field being accessed: ${fieldAccess.className}.${fieldAccess.fieldName}")
-            if (isRealmModelClass(fieldAccess.enclosingClass) && isModelField(fieldAccess.field)) {
+
+            val fieldAccessCtClass: CtClass? = try { classPool.get(fieldAccess.className) } catch (e: NotFoundException) { null }
+            if (fieldAccessCtClass != null && isRealmModelClass(fieldAccessCtClass) && isModelField(fieldAccess.field)) {
                 logger.debug("        Realm: Manipulating ${ctClass.simpleName}.${behaviour.name}(): ${fieldAccess.fieldName}")
                 logger.debug("        Methods: ${ctClass.declaredMethods}")
+
+                // make sure accessors are added, otherwise javassist will fail with
+                // javassist.CannotCompileException: [source error] realmGet$id() not found in 'foo.Model'
+                addRealmAccessors(fieldAccessCtClass)
```

Comment:
It concerns me that this needed?
`addRealmAccessors` should already be called by `BuildTemplate.transformModelClasses()`, so if they are missing it seems to indicate we have other flaws in the transformer logic.

The thinking behind the implementation was:
1) For a full build, all model classes will correctly be modified.
2) For a incremental build, the model class might not be modified, but the accessors would have been added when the model class was compiled for the first time.

Reply:
Also won't this create lots of duplicate method definitions as you are adding Realm accessors for every field access?

Reply:
> addRealmAccessors should already be called by BuildTemplate.transformModelClasses(),

I think that if the model didn't change, then the transformer will not add the model as part of the classes to be modified, this implies that `outputClassNames` will not contain your model so `transformModelClasses` will not generate the accessors again 


The implementation of [addRealmAccessors](https://github.com/realm/realm-java/blob/8aac7d7dab0ddf35624ca2af7aa987995fcceb1e/realm-transformer/src/main/kotlin/io/realm/transformer/ByteCodeModifier.kt#L45) do check if the accessor was already added, so there shouldn't be a duplicate issue

Reply:
I did a little debugging because the above explanation seemed wrong somehow. 
It turns out that on subsequent incremental builds you don't see the transformed output of earlier builds but only the original class without any transforms applied. This is the reason we artificially need to add the accessor again, even though it was already done on a previous build.

You can verify this by first doing a full build (which should add the accessors to all model classes) and then do an incremental build which then fails to find the accessor method.

I also verified it by looking at what classes gets passed to the transformer:

```
// This is the directory containing the non-transformed classes
09:50:12.930 [DEBUG] [realm-logger] Adding: /Users/cm/RealmCustomerProjects/RealmErrorRecreate/app/build/intermediates/classes/debug
```

It is rather annoying we have to add the accessors twice, but the accessors are not written to disk the 2nd time around as the model class is only part of the referenced input.

So, for now, the above solution seems to be the right one, but I'll reach out to Google to get clarification on whether or not this is intended behaviour because somehow it doesn't feel completely right to me.